pkg/thanos/operator.go

```diff
@@ -794,3 +801,9 @@ func newTLSAssetSecret(tr *monitoringv1.ThanosRuler, config Config) *v1.Secret {
 
 	return s
 }
+
+func makeSelectorLabels(name string) map[string]string {
```

Comment:
I would include the labels from here and modify makeStatefulSetSpec() to call makeSelectorLabels() so we have one source of truth:

https://github.com/prometheus-operator/prometheus-operator/blob/838b6a701c0cbd5ff843f7cbcfef7546b4c50eb1/pkg/thanos/statefulset.go#L381-L384